source/extensions/filters/http/jwt_authn/filter_factory.cc

```diff
@@ -46,8 +46,18 @@ FilterFactory::createFilterFactoryFromProtoTyped(const JwtAuthentication& proto_
   validateJwtConfig(proto_config);
   auto filter_config = std::make_shared<FilterConfig>(proto_config, prefix, context);
   return [filter_config](Http::FilterChainFactoryCallbacks& callbacks) -> void {
+    std::vector<AsyncMatcherSharedPtr> rule_matchers;
+    const auto& proto_config = filter_config->getProtoConfig();
+    for (const auto& it : proto_config.rules()) {
```

Comment:
This is the part I don't like.  We are creating all these matchers and verifiers for each request.   This callback will be called for each request.  

We need to come up a way to only create them on the config time.  Not at request processing time.